lib/sentry/client_report.ex

```diff
@@ -0,0 +1,128 @@
+defmodule Sentry.ClientReport do
+  @moduledoc """
+  **TODO** add proper doc for module
+
+  See <https://develop.sentry.dev/sdk/client-reports/>.
+  """
+
+  @moduledoc since: "10.0.0"
+
+  @typedoc since: "10.0.0"
+
+  use GenServer
+  alias Sentry.Client
+
+  @typedoc """
+  The possible reasons of the discarded event.
+  """
+  @type reasons() ::
+          :ratelimit_backoff
+          | :queue_overflow
+          | :cache_overflow
+          | :network_error
+          | :sample_rate
+          | :before_send
+          | :event_processor
+          | :insufficient_data
+          | :backpressure
+
+  @typedoc """
+  The struct for a **client report** interface.
+  """
+  @type t() :: %__MODULE__{
+          timestamp: String.t() | number(),
+          discarded_events:
+            list(%{reason: reasons(), category: String.t(), quantity: pos_integer()})
+        }
+
+  defstruct [:timestamp, :discarded_events]
+
+  @client_report_reasons [
+    :ratelimit_backoff,
+    :queue_overflow,
+    :cache_overflow,
+    :network_error,
+    :sample_rate,
+    :before_send,
+    :event_processor,
+    :insufficient_data,
+    :backpressure
+  ]
+
+  @send_interval 5000
+
+  @spec start_link(keyword()) :: GenServer.on_start()
+  def start_link(_) do
```

Comment:
To make sure we don't pass in unused options:

```suggestion
  def start_link([]) do
```